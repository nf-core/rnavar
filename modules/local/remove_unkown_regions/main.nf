process REMOVE_UNKNOWN_REGIONS {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/python:3.8.3'
        : 'biocontainers/python:3.8.3'}"

    input:
    tuple val(meta), path(bed)
    tuple val(meta2), path(dict)

    output:
    tuple val(meta), path('*.bed'), emit: bed
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    remove_unknown_regions.py ${bed} ${dict} ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
