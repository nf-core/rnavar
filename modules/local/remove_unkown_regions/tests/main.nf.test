nextflow_process {

    name "Test Process REMOVE_UNKNOWN_REGIONS"
    script "../main.nf"
    process "REMOVE_UNKNOWN_REGIONS"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = Channel.value([
                    [id:'exon_bed'],
                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:'dict'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/genome/genome.dict', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should run without failures - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.value([
                    [id:'exon_bed'],
                    file(params.modules_testdata_base_path + '/genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:'dict'],
                    file(params.pipelines_testdata_base_path + '/genomics/homo_sapiens/genome/genome.dict', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
