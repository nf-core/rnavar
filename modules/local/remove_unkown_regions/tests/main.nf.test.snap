{
    "Should run without failures - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "exon_bed"
                        },
                        "exon_bed.exon.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4e809973b78e4e60b7134a81a1eb592b"
                ],
                "bed": [
                    [
                        {
                            "id": "exon_bed"
                        },
                        "exon_bed.exon.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4e809973b78e4e60b7134a81a1eb592b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-05T13:57:48.655747686"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "exon_bed"
                        },
                        "exon_bed.exon.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4e809973b78e4e60b7134a81a1eb592b"
                ],
                "bed": [
                    [
                        {
                            "id": "exon_bed"
                        },
                        "exon_bed.exon.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4e809973b78e4e60b7134a81a1eb592b"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-05T13:57:35.622566479"
    }
}